backend/src/main/java/com/swe573/socialhub/controller/ServiceController.java

```diff
@@ -101,4 +102,25 @@ public void Complete(Principal principal, @PathVariable Long serviceId) {
             throw new ResponseStatusException(HttpStatus.BAD_REQUEST, e.getLocalizedMessage());
         }
     }
+
+    @GetMapping("/flag/{serviceId}")
+    public ResponseEntity<Flag> flagService(Principal principal, @PathVariable Long serviceId) {
+        try {
+            Flag response = serviceService.flagService(principal, serviceId);
+            return ResponseEntity.ok(response);
+        }
+        catch (Exception e) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, e.getLocalizedMessage());
+        }
+    }
+
+    @GetMapping("/flag/control/{serviceId}")
```

Comment:
we can check if a flag exists or not via a GET request

Reply:
done :) 5bb20c01d3d0c77d69ff3b54967f64e53130a6f0